c2rust-ast-exporter/src/AstExporter.cpp

```diff
@@ -2729,9 +2728,14 @@ Outputs process(int argc, const char *argv[], int *result) {
 #endif
 
     // the logic below assumes we're only translating one source file
-    assert(OptionsParser.getSourcePathList().size() - 1 ==
-               source_path_count++ &&
-           "Expected exactly one source path");
+    static size_t source_path_count = 0;
+    source_path_count++;
+    const size_t num_sources = OptionsParser.getSourcePathList().size();
+    assert(
+        (num_sources == 1 // newer clang versions
```

Comment:
could we gate this by clang major version similar to a few lines above?

Reply:
I was thinking about this, but I thought it would be better to have the check based more on a feature test, in case anything else ever changes.  Doing it by clang version seemed more brittle to me, though I think it's an easy enough change.  I'd have to check if it was changed in clang 13 or 14, though.

Reply:
if this fix is specifically a result of differences between clang versions, i think it'd make sense but i'm not sure i understand your point about a feature test. could you explain what you mean?

Reply:
By feature test, I meant something like a JS feature test testing for a feature you want to use/polyfill, rather than checking the browser version.  That is, it's usually better to test for the feature you actually care about (in this case if `OptionsParser.getSourcePathList()` accumulates all the paths across multiple calls) instead of testing indirectly through a version check that may not always be tied to the exact feature.  It could, for example, change in a different way or back in a later clang version.

So we check that either the paths accumulate by the `num_sources == source_path_count` check, and if that's not the case, we can check `num_sources == 1` instead.  That way we don't need to depend on clang version checks where we don't need to, which aren't inherently tied to what we really want to check.  Does that make more sense?

Reply:
Yes that makes sense to me. Is it possible for this assertion to evaluate to true for either set of clang versions where it otherwise wouldn't with one of the two clauses compiled out? I.e. is the feature still meant to work when the clang version is new but only the second clause is true, or if it's old and only the first is true? Or does it not make a difference?

Reply:
Sorry, I'm not exactly sure what you're saying here.  The assertion logic should work regardless of clang version.  A valid (non-null) source path is always chosen and it's always the last one that is chosen.

Reply:
It's alright. It doesn't seem like the scenario I'm thinking about can happen anyway.